### PR TITLE
✨ table methods

### PR DESCRIPTION
**Pull request by rijkvanzanten**
Knex doesn't support selecting from `information_schema` with its api at this point in time. 
Raw queries can't be done on `information_schema` as well, because the table is set on initialization. This limits the data which can be returned in `getTable()` at the moment. 

I will keep looking for a solution

### CODE REVIEW

File: src/local-methods/tables.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+   * Get all table names
+   * @param {Object} params options
+   */
+  getTables(params = {}) {
+    return new Promise((reject, resolve) => {
+      /**
+       * Knex can't access information_schema since it's bound to the
+       *   directus table
+       */
+      this.knex('directus_privileges')
```
Comment (WellingGuzman) [inline src/local-methods/tables.js]: This doesn't provide `directus_tables` inforamation, also `information_schema` is a database, so you would have to use `information_schema.tables`, this table has all the tables from all databases  in mysql.

Read more here: https://dev.mysql.com/doc/refman/5.7/en/information-schema.html

**Comment by rijkvanzanten** [comment]
[`index.js`](https://github.com/directus/directus-sdk-javascript/pull/7/files#diff-168726dbe96b3ce427e7fedce31bb0bc) is the babel-transpiled bundle which npm serves when `require()`ing the module. It has to be checked in git to be able to publish it to npm..

 The actual source code doesn't use one-letter variables 😉

File: src/local-methods/tables.js (hunk 1 above)
Comment (rijkvanzanten) [inline src/local-methods/tables.js]: I could merge in info from `directus_tables`, but I don't have access to the `information_schema` from within the knex instance.. I'm still looking for a way to access this table without having to hack my way around knex.

Comment (WellingGuzman) [inline src/local-methods/tables.js]: According to their documentation this should work:

```js
knex
    .select()
    .from('information_schema.tables')
    .where('table_schema', 'directus')
    .then(function (a) {
        console.log(a);
    })
    .catch(function (err) {
       console.log(err);
    });
```

or:

```js
knex
    .withSchema('information_schema')
    .select()
    .from('tables')
    .where('table_schema', 'directus')
    .then(function (a) {
        console.log(a);
    })
    .catch(function (err) {
       console.log(err);
    });
```
Or using raw query.

I haven't tested it, but either way you should have permission to do it, if the user you are using to connect to the db has enough permissions.

**Comment by WellingGuzman** [comment]
Oh JavaScript and its wonder, I thought you liked see compiled/obfuscated code, I apologies for my ignorance, I do not have much experiences of any of the new javascript fancy stuff, such as babel.

Keep it up! 👍

File: src/local-methods/tables.js (hunk 1 above)
Comment (rijkvanzanten) [inline src/local-methods/tables.js]: I've tried using the raw query but it was trying to perform the query on the `directus` database. I didn't know it was possible to select the table inside the `from` method. I will try it out soon!

**Comment by rijkvanzanten** [comment]
`2.0.0` uses a new code setup. These methods will be re-implemented in this new fashion

**Pull request closed without merging**